src/chapters/errors-and-debugging/categories-of-errors.rst

```diff
@@ -0,0 +1,146 @@
+
+Runtime errors occur during the execution phase of a program. This means that we will only encounter a runtime error once the syntax of our program has been validated. We will only see runtime errors after we have fixed any syntax errors. 
+
+A common runtime error occurs when we try to use a variable that has not been created yet. This can happen if you misspell the name of a variable, as the following example shows.
+
+.. admonition:: Example
+
+   .. sourcecode:: js
+      :linenos:
+   
+      let firstName = "Jack";
+      console.log(firstname);
+
+   **Output**
+
+   ::
+
+      ReferenceError: firstname is not defined
+         at evalmachine.<anonymous>:2:13
+         at Script.runInContext (vm.js:107:20)
+         at Object.runInContext (vm.js:285:6)
+         at evaluate (/run_dir/repl.js:133:14)
+         at ReadStream.<anonymous> (/run_dir/repl.js:116:5)
+         at ReadStream.emit (events.js:189:13)
+         at addChunk (_stream_readable.js:284:12)
+         at readableAddChunk (_stream_readable.js:265:11)
+         at ReadStream.Readable.push (_stream_readable.js:220:10)
+         at lazyFs.read (internal/fs/streams.js:181:12)
+
+The syntax of our program is correct, but when the program executes an error occurs when it reaches line 2. We attempt to print the value of the variable ``firstname``, but such a variable does not exist.
+
+Logic Errors
+------------
+
+.. index::
+   single: error; logic
+
+The third type of error is the **logic error**. If there is a logic error in your program, it will run successfully in the sense that the computer will not generate any error messages. However, the program will not work as intended.
+
+With a logic error, the problem is that the program you wrote is not the program you wanted to write. For example, say you were writing a program to calculate your daily earnings based on your weekly salary. You might write a program like the following:
+
+.. admonition:: Example
+   
+   .. sourcecode:: js
+
+      let weeklyPay = 600;
+
+      let dailyEarnings = weeklyPay / 7;
+      console.log(dailyEarnings);
+
+   **Output**
+
+   ::
+
+      85.71428571428571
+
+The result surprises you because you thought you were making at least $100 per day (you work Monday thru Friday). According to this program, though, you are making about $85 per day. The error is a logic one because you divided your weekly pay by 7. It would have been more accurate to divide your weekly pay by 5, since that is how many days a week you are actually working. 
+
+Identifying logic errors can be tricky because it requires us to work backward by looking at the output of the program and trying to figure out what it is doing. There are no error messages to help us identify the issue.
```

Comment:
Wording: "Identifying logic errors can be tricky because unlike syntax and runtime problems, there are no error messages to help us identify the issue.  We must examine the output of the program and work backward to figure out what it is doing wrong."